### PR TITLE
[feed] migrate to bnd

### PR DESCRIPTION
**Pull request by J-N-K**
This is really a mess.

Parts of the libraries in `org.openhab.io.transport.feed` are non-OSGi-bundles. I orginally wrapped them in the feature but that did not help for the itests because bnd can't do that. Therefore I had to include the library to im- and export the packages (which needs #5448).

Then an implementation of `org.osgi.service.http.HttpService` is needed for the tests to run. I don't know why the resolver did not add that. After adding that ('org.apache.felix.http.jetty`). Now one tests runs, then it hangs forever. I'm lost now.

### CODE REVIEW

**Comment by J-N-K** [comment]
@maggu2810, do you have an idea why the itest runs, finishes and then hangs forever?

**Comment by maggu2810** [comment]
No idea without checking it out and run / debug it.

Have you tried using the Eclipse IDE to run the tests inside and debug it?

**Comment by J-N-K** [comment]
```
! could not resolve the bundles: [org.openhab.binding.feed.tests-2.5.0.201904191716 org.osgi.framework.BundleException: Invalid operation on a fragment.
]
```
Unfortunately I have no idea how to find out what is missing. The bndrun-file is the one generated by the resolver. In `org.openhab.binding.systeminfo` I found (with a lot of luck) that an optional dependency to from an embedded library was the problem. However, we don't have embedded libraries here.

**Comment by J-N-K** [comment]
@openhab/2-x-add-ons-maintainers, why do we have a transport bundle here?  IMO that is very specific to this binding (and consists only of dependencies).

**Comment by kaikreuzer** [comment]
@J-N-K There was once a plan to also implement a feed action/transformation/whatever (I actually can't remember...), so the shared code was put in a separate bundle. As we ended up with only the binding, I think it would be fine to reintegrate this code into the binding.

**Comment by J-N-K** [comment]
Finally I managed to get the tests working. For some unknown reason it did not work with ROME 1.7.4, I had to upgrade to 1.12.0 and add the dependencies to the bindings pom. Since it was unused now, I removed the `io.transport.feed`.

Most difficult to figure out was adding the `HttpService` at port 9090 to the tests.

File: CODEOWNERS [hunk 1]
```diff
@@ -7,8 +7,6 @@
 # Add-on maintainers:
 /addons/binding/org.openhab.binding.enocean/ @fruggy83
 /addons/binding/org.openhab.binding.exec/ @kgoderis
-/addons/binding/org.openhab.binding.feed/ @svilenvul
-/addons/binding/org.openhab.binding.feed.test/ @svilenvul
```
Comment (martinvw) [inline CODEOWNERS]: This one is missing in the CODEOWNERS file

Comment (J-N-K) [reply CODEOWNERS]: fixed

**Comment by maggu2810** [comment]
> Finally I managed to get the tests working.

Great! It has not been possible for me to find any spare time to have a look at. Sorry.

**Comment by davidgraeff** [comment]
Can you resolve the conflicts? Thanks

**Comment by J-N-K** [comment]
fixed

**Comment by davidgraeff** [comment]
by merging the openhab cloud PR this one gets a conflict again, sry

**Comment by J-N-K** [comment]
solved

**Pull request merged**